packages/aws-cdk/lib/api/logs/logs-monitor.ts

```diff
@@ -0,0 +1,209 @@
+import * as util from 'util';
+import * as cxapi from '@aws-cdk/cx-api';
+import * as colors from 'colors/safe';
+import { print, error } from '../../logging';
+import { flatten } from '../../util/arrays';
+import { ISDK } from '../aws-auth';
+
+// how often we should read events from CloudWatchLogs
+const SLEEP = 2_000;
```

Comment:
What do you mean? Maybe I should reword this comment, it is the amount of time we wait after processing events before we start processing new events. So we:

1.  read all events starting with `startTime` for all log groups
2. print all the events
3. SLEEP
4. start back at 1

Reply:
I'm just a little concerned about overwhelming CloudWatch Logs with `filterLogEvents` calls, and getting throttled.

Typically, in a throttling event you do an exponential backoff to avoid overloading the backend service. I suppose there's an argument to be made that the SDK's retry settings would take care of this for us... but if that's your argument I'd like to see that explicitly stated somewhere.

Reply:
I added a comment here to note that sleep is also how we are handling backoff. Now that we are not using `nextToken` we will only make 1 request per log group. If any of the requests are throttled the SDKs builtin retry/backoff will take effect. If that doesn't resolve it, it will throw an error, wait for `SLEEP` and then try again.